include/mem.h

```diff
@@ -41,136 +41,172 @@ void MEM_A20_Enable(bool enable);
 }
-static INLINE Bit16u phys_readw(PhysPt addr){
-	return host_readw(MemBase+addr);
+static inline uint16_t phys_readw(PhysPt addr)
+{
+	return host_readw(MemBase + addr);
 }
-static INLINE Bit32u phys_readd(PhysPt addr){
-	return host_readd(MemBase+addr);
+static inline uint32_t phys_readd(PhysPt addr)
+{
+	return host_readd(MemBase + addr);
 }
 
 /* These don't check for alignment, better be sure it's correct */
 
-void MEM_BlockWrite(PhysPt pt,void const * const data,Bitu size);
-void MEM_BlockRead(PhysPt pt,void * data,Bitu size);
-void MEM_BlockCopy(PhysPt dest,PhysPt src,Bitu size);
-void MEM_StrCopy(PhysPt pt,char * data,Bitu size);
+void MEM_BlockWrite(PhysPt pt, void const *const data, Bitu size);
+void MEM_BlockRead(PhysPt pt, void *data, Bitu size);
+void MEM_BlockCopy(PhysPt dest, PhysPt src, Bitu size);
+void MEM_StrCopy(PhysPt pt, char *data, Bitu size);
 
-void mem_memcpy(PhysPt dest,PhysPt src,Bitu size);
+void mem_memcpy(PhysPt dest, PhysPt src, Bitu size);
 Bitu mem_strlen(PhysPt pt);
-void mem_strcpy(PhysPt dest,PhysPt src);
+void mem_strcpy(PhysPt dest, PhysPt src);
 
-/* The folowing functions are all shortcuts to the above functions using physical addressing */
+/* The folowing functions are all shortcuts to the above functions using
+ * physical addressing */
 
-static INLINE Bit8u real_readb(Bit16u seg,Bit16u off) {
-	return mem_readb((seg<<4)+off);
+static inline uint8_t real_readb(uint16_t seg, uint16_t off)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	return mem_readb(base + off);
 }
-static INLINE Bit16u real_readw(Bit16u seg,Bit16u off) {
-	return mem_readw((seg<<4)+off);
+static inline uint16_t real_readw(uint16_t seg, uint16_t off)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	return mem_readw(base + off);
 }
-static INLINE Bit32u real_readd(Bit16u seg,Bit16u off) {
-	return mem_readd((seg<<4)+off);
+static inline uint32_t real_readd(uint16_t seg, uint16_t off)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	return mem_readd(base + off);
 }
 
-static INLINE void real_writeb(Bit16u seg,Bit16u off,Bit8u val) {
-	mem_writeb(((seg<<4)+off),val);
+static inline void real_writeb(uint16_t seg, uint16_t off, uint8_t val)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	mem_writeb(base + off, val);
 }
-static INLINE void real_writew(Bit16u seg,Bit16u off,Bit16u val) {
-	mem_writew(((seg<<4)+off),val);
+static inline void real_writew(uint16_t seg, uint16_t off, uint16_t val)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	mem_writew(base + off, val);
 }
-static INLINE void real_writed(Bit16u seg,Bit16u off,Bit32u val) {
-	mem_writed(((seg<<4)+off),val);
+static inline void real_writed(uint16_t seg, uint16_t off, uint32_t val)
+{
+	const auto base = static_cast<uint32_t>(seg << 4);
+	mem_writed(base + off, val);
 }
 
-
-static INLINE Bit16u RealSeg(RealPt pt) {
-	return (Bit16u)(pt>>16);
+static inline uint16_t RealSeg(RealPt pt)
+{
+	return (uint16_t)(pt >> 16);
```

Comment:
`static_cast`?

Reply:
Was able to eliminate the cast entirely, given the shift takes the 32-bit `pt` value down to 16-bits.